File: projectflutter/lib/data/exercise/request/exercise_session_batch_request.dart
import 'package:projectflutter/data/exercise/request/exercise_session_request.dart';

class ExerciseSessionBatchRequest {
  final String userId;
  final int subCategoryId;
  final int resetBatch;
  final List<ExerciseSessionRequest> sessions;

  ExerciseSessionBatchRequest({
    required this.userId,
    required this.subCategoryId,
    required this.resetBatch,
    required this.sessions,
  });

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'subCategoryId': subCategoryId,
      'resetBatch': resetBatch,
      'sessions': sessions.map((e) => e.toJson()).toList(),
    };
  }
}

File: finalProject_backend/src/main/java/aptech/finalproject/payload/exercise/ExerciseSessionBatchRequest.java
package aptech.finalproject.payload.exercise;

import java.util.List;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ExerciseSessionBatchRequest {
    private List<ExerciseSessionRequest> sessions;
    private String userId;
    private Integer resetBatch;
    private Integer subCategoryId;
}